### ISSUE TITLE
WireGuard support

### ISSUE THREAD

**Issue by dhess** [LnL7/nix-darwin#79]
wireguard-go now supports macOS. See https://lists.zx2c4.com/pipermail/wireguard/2018-May/002854.html

It might be nice to have some configuration for this, similar to what the NixOS module provides.

**Comment by kirelagin** [comment]
I am definitely interesting in doing this, but I am not sure when I’ll have time (might be today, might be in a week), so if someone wants to work on this, please, do.

**Comment by LoganBarnett** [comment]
I was able to setup a working Wireguard setup on `nix-darwin` using `wg-quick` declarative options (seen here: https://github.com/LnL7/nix-darwin/blob/ae406c04577ff9a64087018c79b4fdc02468c87c/modules/services/wg-quick.nix).  Sounds like we can close this :)

**Comment by thomasjm** [comment]
FWIW, I think the `wg-quick` options should be edited to add `/bin/wait4path` call, like some services do. I had to add the following for an interface named "wg0":

```nix
  launchd.daemons.wg-quick-wg0.serviceConfig.ProgramArguments = lib.mkForce [
    "/bin/sh"
    "-c"
    "/bin/wait4path /nix/store &amp;&amp; exec ${pkgs.wireguard-tools}/bin/wg-quick up wg0"
  ];
```

I think this is needed because `/nix/store` gets mounted by a launchd service on macOS? It's possible this only applies to me because I have /nix/store on an external disk though.

**Comment by Samasaur1** [comment]
Yeah, you will need `/bin/wait4path`. I'd probably recommend setting `script` or `command` instead of `serviceOptions.ProgramArguments` because it will automatically prefix `/bin/wait4path` (and if we change how launch daemons work (https://github.com/nix-darwin/nix-darwin/issues/1219#issuecomment-2573095530), it will stay up to date)
